### ISSUE TITLE
`lcnt` crashes occasionally when analyzing data

### ISSUE THREAD

**Issue by mkuratczyk** [erlang/otp#4641]
**Describe the bug**
`lcnt` crashed twice for me today - in one case when running `lcnt:conflicts()` and another time during `lcnt:inspect(mnesia_locker)`. The crashes seem very similar so I'm reporting them as a single issue.

**To Reproduce**
Unfortunately I was unable to deterministically reproduce the problem. The crash happened when analyzing a RabbitMQ cluster with high lock contention but most `lcnt` calls completed successfully (including `lcnt:conflicts/0` and `lcnt:inspect/1`) - the crash only happened twice.

**Affected versions**
23.2.7 with `--enable-lock-counter`

**Additional context**
Attaching the output of the two crashes.

[lcnt-crash.log.gz](https://github.com/erlang/otp/files/6158188/lcnt-crash.log.gz)
[lcnt-crash2.log.gz](https://github.com/erlang/otp/files/6158189/lcnt-crash2.log.gz)

**Comment by garazdawi** [comment]
Thanks for reporting! That error happens when the group leader of the process calling `lcnt` has terminated. How are you calling `lcnt`?

**Comment by mkuratczyk** [comment]
I had a remote shell connected to out one the RabbitMQ nodes, ran `lcnt:clear/0`, `lcnt:collect/0`, `lcnt:conflicts/0` and a bunch of `lcnt:inspect/1` when one of those conflicts/inspect calls crashed. Nothing significant changed between the time the commands worked and the execution that crashed - I was in the same shell, connected to the same node, just running yet another command.

**Comment by garazdawi** [comment]
Could it be that you re-attached to the node and ran lcnt again?

**Comment by ranjanified** [comment]
Only guessing....
Could it be that he re-attached, but he didn't notice anything ??

**Comment by mkuratczyk** [comment]
Sorry, you are right - my previous comment was an incorrect recollection of yesterday's events. So I must have had a remote shell when I ran the successful commands and then opened another remote shell where it crashed. This is in my terminal history:

```
erl -emu_type lcnt -sname debug2 -hidden -remsh rabbit-2
Erlang/OTP 23 [erts-11.1.8] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [lock-counting]

Eshell V11.1.8  (abort with ^G)
(rabbit-2@mkuratczyk-a01)1> lcnt:swap_pid_keys().
ok
(rabbit-2@mkuratczyk-a01)2> lcnt:conflicts().
=ERROR REPORT==== 17-Mar-2021::17:47:39.505059 ===
```

```
$ erl -sname debug2 -hidden -remsh rabbit-1

Erlang/OTP 23 [erts-11.1.8] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]

Eshell V11.1.8  (abort with ^G)
(rabbit-1@mkuratczyk-a01)1> lcnt:inspect(mnesia_locker).
=ERROR REPORT==== 17-Mar-2021::14:06:39.444007 ===
```

**Comment by garazdawi** [comment]
No, it is not. But now we know what the problem is. `lcnt` starts a server that gets the `group_leader` of the first `-remsh`. Then when that `-remsh` terminates, the `group_leader` and `lcnt` server remains and the next `-remsh` tries to execute a command which makes the `lcnt` server print to the old `group_leader` and crashes.

The solution here is to propagate the group leader through the gen_server:calls being done to the server.

A PR fixing this would be most welcome. If not, we will fix this problem as time allows.

**Comment by mkuratczyk** [comment]
Great, I have an obvious workaround (just using one shell) and I will look into potentially submitting a PR. Thank you for such a quick response!
